core/shared/src/main/scala/scalaz/zio/internal/FiberContext.scala

```diff
@@ -507,22 +478,30 @@ private[zio] final class FiberContext[E, A](
         if (!state.compareAndSet(oldState, Executing(true, status, observers))) kill0(k)
         else Async.later
 
-      case Done(e) => Async.now(ExitResult.succeeded(e))
+      case Done(e) => Async.now(IO.now(e))
     }
   }
 
+  private[this] final def observe0(
+    k: Callback[Nothing, ExitResult[E, A]]
+  ): Async[Nothing, ExitResult[E, A]] =
+    register0(k) match {
+      case null => Async.later
+      case x    => Async.now(IO.now(x))
```

Comment:
What about:
```scala
object Async {
  def now[A](value: A): Async[Nothing, A] = Now(IO.now(a))
  def fail[E](error: E): Async[E, Nothing] = Now(IO.fail(e))
}
```

Reply:
Then we'd need to add `success` / `interrupt`, `fail0`, etc., and it would start to look like `IO`.

Reply:
Yeah, fair enough 👍